GoodMorning/GoodMorning/Domain/UseCases/FetchCurrentWeatherUseCase.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  FetchCurrentWeatherUseCase.swift
+//  GoodMorning
+//
+//  Created by Miro on 11/2/23.
+//
+
+import Foundation
+
+final class FetchCurrentWeatherUseCase {
+
+    private let weatherRepository: WeatherRepository
+    private let locationRepository: CurrentLocationRepository
+
+    init(weatherRepository: WeatherRepository, locationRepository: CurrentLocationRepository) {
+        self.weatherRepository = weatherRepository
+        self.locationRepository = locationRepository
+    }
+
+    func execute() async throws -> CurrentWeather {
+
+        let locationResult = locationRepository.fetchCurrentLocation()
+
+        switch locationResult {
+        case .success(let coordinate):
+            let currentWeatherResult = try await weatherRepository.fetchCurrentWeather(
+                in: coordinate ?? Coordinate(longitude: "0", latitude: "0")
+            )
+
+            switch currentWeatherResult {
+            case .success(let weather):
+                return weather
+            case .failure(let networkError):
+                throw networkError
+            }
```

Comment:
중첩 switch 문에 대해서도 이야기 하셨었는데, 저도 중첩 switch문을 보니 코드를 한번에 이해하기가 쉽지 않다고 느꼈어요. 혹시 아래와 같은 방식으로 메서드를 나눠 보는건 어떨까요?!
```suggestion
 func execute() async throws -> CurrentWeather {

        guard let locationResult = locationRepository.fetchCurrentLocation() else {
            throw LocationError.unknowned
        }

        switch locationResult {
        case .success(let coordinate):
            guard let coordinate else {
                // location이 nil일 때,,
                throw LocationError.unknowned
            }
            return try await self.fetchCurrentWeather(coordinate)
        case .failure(let error):
            throw error
        }
    }

    private func fetchCurrentWeather(_ coordinate: Coordinate) async throws -> CurrentWeather {

        let currentWeatherResult = try await weatherRepository.fetchCurrentWeather(
            in: coordinate
        )

        switch currentWeatherResult {
        case .success(let weather):
            return weather
        case .failure(let networkError):
            throw networkError
        }
    }
```

Reply:
앗 너무 좋네요!!ㅎㅎ 이렇게 간단하게 함수로 분리할 생각을 못했네요..ㅎㅎ👍